src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -3404,6 +3403,11 @@ private unsafe void ParseRemaining()
                 }
 
                 length = (ushort)_string.Length;
+                // We need to be sure that there isn't ? after path
```

Comment:
Because if after we've other fragment it's ok keep it and below the len passed to canonical will lead to "non canonical"(space must be escaped if is between path and query for example ".../path ?param=value" but it ok to remove for ".../path ")

Reply:
My explanation is very poor :pensive: sorry. I'll try be more clear:
```cs
new object[] { "http://www.contoso.com/path ?name ", false },
new object[] { "http://www.contoso.com/path ?\u00E4 ", false }
```

Here on Iri we are analyzing Path fragment. So _string will be "http://www.contoso.com/path " if we trim space below on [CheckCanonical](https://github.com/dotnet/corefx/blob/5216596a25c587ee2d9ce89e9d0d26b781742267/src/System.Private.Uri/src/System/Uri.cs#L3421) won't result in "non canonical"(lead to test fail...but Iri isn't canonical space should be escaped). So my idea is to verify that we "cut" only if we are analyzing complete uri(no more fragment after current in this case Path...so we don't have ? after)and not only that fragment type.
Feel free to change/enrich comments.

Reply:
Would
```
// We need to be sure that there isn't a '?' separated from the path by spaces.
```
be accurate?

Reply:
great!